test/e2e/rule_test.go

```diff
@@ -62,19 +70,321 @@ func createRuleFiles(t *testing.T, dir string) {
+	// HTTPS with authentication.
+	handler2 := newMockAlertmanager("/api/v1/alerts", "secret")
+	srv2 := httptest.NewTLSServer(handler2)
+	defer srv2.Close()
+
+	// Write the server's certificate to disk for the alerting configuration.
+	tlsDir, err := ioutil.TempDir("", "tls")
+	defer os.RemoveAll(tlsDir)
+	testutil.Ok(t, err)
+	var out bytes.Buffer
+	err = pem.Encode(&out, &pem.Block{Type: "CERTIFICATE", Bytes: srv2.TLS.Certificates[0].Certificate[0]})
+	testutil.Ok(t, err)
+	caFile := filepath.Join(tlsDir, "ca.crt")
+	err = ioutil.WriteFile(caFile, out.Bytes(), 0640)
+	testutil.Ok(t, err)
+
+	amCfg := serializeAlertingConfiguration(
+		t,
+		alert.AlertmanagerConfig{
+			StaticAddresses: []string{srv1.Listener.Addr().String()},
+			Scheme:          "http",
+			Timeout:         model.Duration(time.Second),
+			PathPrefix:      "/prefix/",
+		},
+		alert.AlertmanagerConfig{
+			HTTPClientConfig: alert.HTTPClientConfig{
+				TLSConfig: alert.TLSConfig{
+					CAFile: caFile,
+				},
+				BearerToken: "secret",
+			},
+			StaticAddresses: []string{srv2.Listener.Addr().String()},
+			Scheme:          "https",
+			Timeout:         model.Duration(time.Second),
+		},
+	)
+
+	rulesDir, err := ioutil.TempDir("", "rules")
+	defer os.RemoveAll(rulesDir)
+	testutil.Ok(t, err)
+	createRuleFiles(t, rulesDir)
+
+	qAddr := a.New()
+	r := rule(a.New(), a.New(), rulesDir, amCfg, []address{qAddr}, nil)
+	q := querier(qAddr, a.New(), []address{r.GRPC}, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	exit, err := e2eSpinup(t, ctx, q, r)
+	if err != nil {
+		t.Errorf("spinup failed: %v", err)
+		cancel()
+		return
+	}
+
+	defer func() {
+		cancel()
+		<-exit
+	}()
+
+	testutil.Ok(t, runutil.Retry(5*time.Second, ctx.Done(), func() (err error) {
+		select {
+		case <-exit:
+			cancel()
+			return nil
+		default:
+		}
+
+		for i, am := range []*mockAlertmanager{handler1, handler2} {
+			if len(am.Alerts()) == 0 {
+				return errors.Errorf("no alert received from handler%d, last error: %v", i, am.LastError())
+			}
+		}
+
+		return nil
+	}))
+}
+
+func TestRuleAlertmanagerFileSD(t *testing.T) {
+	a := newLocalAddresser()
+
+	am := alertManager(a.New())
+	amDir, err := ioutil.TempDir("", "am")
+	defer os.RemoveAll(amDir)
+	testutil.Ok(t, err)
+	amCfg := serializeAlertingConfiguration(
+		t,
+		alert.AlertmanagerConfig{
+			FileSDConfigs: []alert.FileSDConfig{
+				alert.FileSDConfig{
+					Files:           []string{filepath.Join(amDir, "*.yaml")},
+					RefreshInterval: model.Duration(time.Hour),
+				},
+			},
+			Scheme:  "http",
+			Timeout: model.Duration(time.Second),
+		},
+	)
+
+	rulesDir, err := ioutil.TempDir("", "rules")
+	defer os.RemoveAll(rulesDir)
+	testutil.Ok(t, err)
+	createRuleFiles(t, rulesDir)
+
+	qAddr := a.New()
+	r := rule(a.New(), a.New(), rulesDir, amCfg, []address{qAddr}, nil)
+	q := querier(qAddr, a.New(), []address{r.GRPC}, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
```

Comment:
`1*time.Minute` This might be not enough for our sometimes slow CI, let's make it 3m

Reply:
still 1m?